### PR TITLE
65 add the question entity

### PR DESCRIPTION
**Pull request by yusufzartit**

### CODE REVIEW

File: src/main/java/com/iotiq/application/service/wikiDataService/QuestionService.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+public class QuestionService {
+    public List<Question> getQuestions(String categoryName, String difficultyLevel, int numberOfQuestions) {
+        List<QuestionDto> questionDtos = WikiDummyDataHolder.questions;
+        return convertQuestionDtosToQuestions(questionDtos);
+    }
+
+    public List<Question> convertQuestionDtosToQuestions(List<QuestionDto> questionDtos) {
+        return questionDtos.stream()
+                .map(dto -> {
+                    Question question = new Question();
```
Comment (oguzIotiq) [inline src/main/java/com/iotiq/application/service/wikiDataService/QuestionService.java:23]: you can add a method to convert a single dto to a single question and use it here (convertQuestionDtoToQuestion)

File: src/main/java/com/iotiq/application/domain/Question.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+public class Question extends AbstractPersistable<UUID> {
+    private String title;
+    private String hint;
+    @Enumerated(EnumType.STRING)
+    private QuestionType questionType;
+    @OneToMany(cascade = CascadeType.ALL)
+    @JoinColumn(name = "question_id")
+    private List<Option> options;
+    private String rightAnswer;
+    private String pointAnswer;
```
[the remark this answers is not in the document]
Comment (yusufzartit) [reply src/main/java/com/iotiq/application/domain/Question.java:24]: Represents the score for the correct answer to the question, and has been renamed correctAnswerPoints.

File: src/main/java/com/iotiq/application/domain/Option.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+import org.springframework.data.jpa.domain.AbstractPersistable;
+
+import java.util.UUID;
+@Entity
+@Setter
+@Getter
+public class Option extends AbstractPersistable<UUID> {
+    private String text;
+    @ManyToOne
+    private Question question;
```
[the remark this answers is not in the document]
Comment (yusufzartit) [reply src/main/java/com/iotiq/application/domain/Option.java:16]: Actually we don't need a Bidirectional relation, it has been replaced with a Unidirectional relation.

File: src/main/java/com/iotiq/application/service/wikiDataService/QuestionService.java (hunk 1 above)
Comment (yusufzartit) [reply src/main/java/com/iotiq/application/service/wikiDataService/QuestionService.java]: It has been fixed.

**Review by oguzIotiq — APPROVED** [review]

**Pull request merged**